src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -456,20 +461,42 @@ public void terminateTask(String taskId, TerminateInfo terminateInfo) {
      * @param initiateTaskRequest Additional data to define how a task should be initiated.
      * @return The updated entity {@link TaskResource}
      */
-    @Transactional(rollbackFor = Exception.class)
     public TaskResource initiateTask(String taskId, InitiateTaskRequest initiateTaskRequest) {
-        TaskResource taskResource = createTaskSkeleton(taskId, initiateTaskRequest);
-        if (canGetDbLock(taskResource)) {
-            taskResource = configureTask(taskResource);
-            taskResource = taskAutoAssignmentService.autoAssignCFTTask(taskResource);
-            updateCftTaskState(taskResource.getTaskId(), taskResource);
+
+        Optional<TaskResource> maybeTaskResource = cftTaskDatabaseService.findByIdOnly(taskId);
+
+        if (maybeTaskResource.isPresent()) {
+            if (UNCONFIGURED.equals(maybeTaskResource.get().getState())) {
+                return runTaskThroughInitiation(taskId);
+            } else {
+                throw new DatabaseConflictException(ErrorMessages.DATABASE_CONFLICT_ERROR);
+            }
+        } else {
+            createTaskSkeletonAndSave(taskId, initiateTaskRequest);
+            return runTaskThroughInitiation(taskId);
         }
-        return cftTaskDatabaseService.saveTask(taskResource);
     }
 
-    private boolean canGetDbLock(TaskResource taskResource) {
-        return cftTaskDatabaseService.saveTask(taskResource).getTaskId()
-            .equals(taskResource.getTaskId());
+    private TaskResource runTaskThroughInitiation(String taskId) {
+        //First lock the task
+        Optional<TaskResource> maybeLockedTaskResource = cftTaskDatabaseService
```

Comment:
this could throw an exception, is it handled in the Advice Controller?

Reply:
At the moment yes it will be mapped to a generic server error

Reply:
+1